### ISSUE TITLE
PyYAML requirements inconsistent

### ISSUE THREAD

**Issue by rjaduthie** [localstack/localstack#1052]
Similar to #488.  Using `localstack-0.8.8`, `Python 3.6.7` and `pipenv 2018.10.13`.

Installing via `pipenv install localstack` there seems to be internal inconsistency in PyYAML requirements.  Localstack wants ==4.2b4, but AWS CLI wants >=3.0 <=3.13.  Everything else seems to be happy with anything.

`pipenv graph` gives (search for PyYAML and pyyaml to find inconsistencies):
```
localstack==0.8.8
  - airspeed [required: ==0.5.5.dev20160812, installed: 0.5.5.dev20160812]
    - cachetools [required: ==0.8.0, installed: 0.8.0]
    - six [required: Any, installed: 1.10.0]
  - awscli [required: >=1.14.18, installed: 1.16.72]
    - botocore [required: ==1.12.62, installed: 1.12.62]
      - docutils [required: >=0.10, installed: 0.14]
      - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
      - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
        - six [required: >=1.5, installed: 1.10.0]
      - urllib3 [required: >=1.20,<1.25, installed: 1.22]
    - colorama [required: >=0.2.5,<=0.3.9, installed: 0.3.9]
    - docutils [required: >=0.10, installed: 0.14]
    - PyYAML [required: >=3.10,<=3.13, installed: 4.2b4]
    - rsa [required: >=3.1.2,<=3.5.0, installed: 3.4.2]
      - pyasn1 [required: >=0.1.3, installed: 0.4.4]
    - s3transfer [required: >=0.1.12,<0.2.0, installed: 0.1.13]
      - botocore [required: >=1.3.0,<2.0.0, installed: 1.12.62]
        - docutils [required: >=0.10, installed: 0.14]
        - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
        - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
          - six [required: >=1.5, installed: 1.10.0]
        - urllib3 [required: >=1.20,<1.25, installed: 1.22]
  - boto [required: ==2.46.1, installed: 2.46.1]
  - boto3 [required: >=1.4.5, installed: 1.9.62]
    - botocore [required: >=1.12.62,<1.13.0, installed: 1.12.62]
      - docutils [required: >=0.10, installed: 0.14]
      - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
      - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
        - six [required: >=1.5, installed: 1.10.0]
      - urllib3 [required: >=1.20,<1.25, installed: 1.22]
    - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
    - s3transfer [required: >=0.1.10,<0.2.0, installed: 0.1.13]
      - botocore [required: >=1.3.0,<2.0.0, installed: 1.12.62]
        - docutils [required: >=0.10, installed: 0.14]
        - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
        - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
          - six [required: >=1.5, installed: 1.10.0]
        - urllib3 [required: >=1.20,<1.25, installed: 1.22]
  - coverage [required: ==4.0.3, installed: 4.0.3]
  - docopt [required: ==0.6.2, installed: 0.6.2]
  - elasticsearch [required: ==6.2.0, installed: 6.2.0]
    - urllib3 [required: >=1.21.1,<1.23, installed: 1.22]
  - flake8 [required: ==3.6.0, installed: 3.6.0]
    - mccabe [required: >=0.6.0,<0.7.0, installed: 0.6.1]
    - pycodestyle [required: >=2.4.0,<2.5.0, installed: 2.4.0]
    - pyflakes [required: >=2.0.0,<2.1.0, installed: 2.0.0]
    - setuptools [required: >=30, installed: 40.6.2]
  - flake8-quotes [required: >=0.11.0, installed: 1.0.0]
    - flake8 [required: Any, installed: 3.6.0]
      - mccabe [required: >=0.6.0,<0.7.0, installed: 0.6.1]
      - pycodestyle [required: >=2.4.0,<2.5.0, installed: 2.4.0]
      - pyflakes [required: >=2.0.0,<2.1.0, installed: 2.0.0]
      - setuptools [required: >=30, installed: 40.6.2]
  - flask [required: ==0.12.3, installed: 0.12.3]
    - click [required: >=2.0, installed: 7.0]
    - itsdangerous [required: >=0.21, installed: 1.1.0]
    - Jinja2 [required: >=2.4, installed: 2.10]
      - MarkupSafe [required: >=0.23, installed: 1.1.0]
    - Werkzeug [required: >=0.7, installed: 0.14.1]
  - flask-cors [required: ==3.0.3, installed: 3.0.3]
    - Flask [required: >=0.9, installed: 0.12.3]
      - click [required: >=2.0, installed: 7.0]
      - itsdangerous [required: >=0.21, installed: 1.1.0]
      - Jinja2 [required: >=2.4, installed: 2.10]
        - MarkupSafe [required: >=0.23, installed: 1.1.0]
      - Werkzeug [required: >=0.7, installed: 0.14.1]
    - Six [required: Any, installed: 1.10.0]
  - flask-swagger [required: ==0.2.12, installed: 0.2.12]
    - Flask [required: >=0.10, installed: 0.12.3]
      - click [required: >=2.0, installed: 7.0]
      - itsdangerous [required: >=0.21, installed: 1.1.0]
      - Jinja2 [required: >=2.4, installed: 2.10]
        - MarkupSafe [required: >=0.23, installed: 1.1.0]
      - Werkzeug [required: >=0.7, installed: 0.14.1]
    - PyYAML [required: >=3.0, installed: 4.2b4]
  - jsonpath-rw [required: ==1.4.0, installed: 1.4.0]
    - decorator [required: Any, installed: 4.3.0]
    - ply [required: Any, installed: 3.11]
    - six [required: Any, installed: 1.10.0]
  - localstack-client [required: ==0.6, installed: 0.6]
    - boto3 [required: Any, installed: 1.9.62]
      - botocore [required: >=1.12.62,<1.13.0, installed: 1.12.62]
        - docutils [required: >=0.10, installed: 0.14]
        - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
        - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
          - six [required: >=1.5, installed: 1.10.0]
        - urllib3 [required: >=1.20,<1.25, installed: 1.22]
      - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
      - s3transfer [required: >=0.1.10,<0.2.0, installed: 0.1.13]
        - botocore [required: >=1.3.0,<2.0.0, installed: 1.12.62]
          - docutils [required: >=0.10, installed: 0.14]
          - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
          - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
            - six [required: >=1.5, installed: 1.10.0]
          - urllib3 [required: >=1.20,<1.25, installed: 1.22]
  - localstack-ext [required: >=0.8.6, installed: 0.8.6]
    - dnslib [required: ==0.9.7, installed: 0.9.7]
    - pyaes [required: ==1.6.0, installed: 1.6.0]
    - pyminifier [required: ==2.1, installed: 2.1]
    - srp-ext [required: ==1.0.7.1, installed: 1.0.7.1]
      - six [required: Any, installed: 1.10.0]
    - warrant-ext [required: >=0.6.1.1, installed: 0.6.1.1]
      - boto3 [required: >=1.4.3, installed: 1.9.62]
        - botocore [required: >=1.12.62,<1.13.0, installed: 1.12.62]
          - docutils [required: >=0.10, installed: 0.14]
          - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
          - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
            - six [required: >=1.5, installed: 1.10.0]
          - urllib3 [required: >=1.20,<1.25, installed: 1.22]
        - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
        - s3transfer [required: >=0.1.10,<0.2.0, installed: 0.1.13]
          - botocore [required: >=1.3.0,<2.0.0, installed: 1.12.62]
            - docutils [required: >=0.10, installed: 0.14]
            - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
            - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
              - six [required: >=1.5, installed: 1.10.0]
            - urllib3 [required: >=1.20,<1.25, installed: 1.22]
      - envs [required: >=0.3.0, installed: 1.2.6]
      - mock [required: >=2.0.0, installed: 2.0.0]
        - pbr [required: >=0.11, installed: 5.1.1]
        - six [required: >=1.9, installed: 1.10.0]
      - python-jose-ext [required: >=1.3.2.4, installed: 1.3.2.4]
        - ecdsa [required: <1.0, installed: 0.13]
        - future [required: <1.0, installed: 0.17.1]
        - pycryptodomex [required: ==3.4.9, installed: 3.4.9]
        - six [required: <2.0, installed: 1.10.0]
      - requests [required: >=2.13.0, installed: 2.20.0]
        - certifi [required: >=2017.4.17, installed: 2018.11.29]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.8, installed: 2.7]
        - urllib3 [required: >=1.21.1,<1.25, installed: 1.22]
  - moto-ext [required: ==1.3.4, installed: 1.3.4]
    - aws-xray-sdk [required: >=0.93,<0.96, installed: 0.95]
      - jsonpickle [required: Any, installed: 1.0]
      - requests [required: Any, installed: 2.20.0]
        - certifi [required: >=2017.4.17, installed: 2018.11.29]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.8, installed: 2.7]
        - urllib3 [required: >=1.21.1,<1.25, installed: 1.22]
      - wrapt [required: Any, installed: 1.10.11]
    - boto [required: >=2.36.0, installed: 2.46.1]
    - boto3 [required: >=1.6.16, installed: 1.9.62]
      - botocore [required: >=1.12.62,<1.13.0, installed: 1.12.62]
        - docutils [required: >=0.10, installed: 0.14]
        - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
        - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
          - six [required: >=1.5, installed: 1.10.0]
        - urllib3 [required: >=1.20,<1.25, installed: 1.22]
      - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
      - s3transfer [required: >=0.1.10,<0.2.0, installed: 0.1.13]
        - botocore [required: >=1.3.0,<2.0.0, installed: 1.12.62]
          - docutils [required: >=0.10, installed: 0.14]
          - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
          - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
            - six [required: >=1.5, installed: 1.10.0]
          - urllib3 [required: >=1.20,<1.25, installed: 1.22]
    - botocore [required: >=1.9.16, installed: 1.12.62]
      - docutils [required: >=0.10, installed: 0.14]
      - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.3]
      - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
        - six [required: >=1.5, installed: 1.10.0]
      - urllib3 [required: >=1.20,<1.25, installed: 1.22]
    - cookies [required: Any, installed: 2.2.1]
    - cryptography [required: >=2.0.0, installed: 2.4.2]
      - asn1crypto [required: >=0.21.0, installed: 0.24.0]
      - cffi [required: >=1.7,!=1.11.3, installed: 1.11.5]
        - pycparser [required: Any, installed: 2.19]
      - idna [required: >=2.1, installed: 2.7]
      - six [required: >=1.4.1, installed: 1.10.0]
    - docker [required: >=2.5.1, installed: 3.6.0]
      - docker-pycreds [required: >=0.3.0, installed: 0.4.0]
        - six [required: >=1.4.0, installed: 1.10.0]
      - requests [required: >=2.14.2,!=2.18.0, installed: 2.20.0]
        - certifi [required: >=2017.4.17, installed: 2018.11.29]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.8, installed: 2.7]
        - urllib3 [required: >=1.21.1,<1.25, installed: 1.22]
      - six [required: >=1.4.0, installed: 1.10.0]
      - websocket-client [required: >=0.32.0, installed: 0.54.0]
        - six [required: Any, installed: 1.10.0]
    - Jinja2 [required: >=2.7.3, installed: 2.10]
      - MarkupSafe [required: >=0.23, installed: 1.1.0]
    - jsondiff [required: ==1.1.1, installed: 1.1.1]
    - mock [required: Any, installed: 2.0.0]
      - pbr [required: >=0.11, installed: 5.1.1]
      - six [required: >=1.9, installed: 1.10.0]
    - pyaml [required: Any, installed: 18.11.0]
      - PyYAML [required: Any, installed: 4.2b4]
    - python-dateutil [required: >=2.1,<3.0.0, installed: 2.7.5]
      - six [required: >=1.5, installed: 1.10.0]
    - python-jose [required: <3.0.0, installed: 2.0.2]
      - ecdsa [required: <1.0, installed: 0.13]
      - future [required: <1.0, installed: 0.17.1]
      - pycryptodome [required: >=3.3.1,<4.0.0, installed: 3.7.2]
      - six [required: <2.0, installed: 1.10.0]
    - pytz [required: Any, installed: 2018.7]
    - requests [required: >=2.5, installed: 2.20.0]
      - certifi [required: >=2017.4.17, installed: 2018.11.29]
      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<2.8, installed: 2.7]
      - urllib3 [required: >=1.21.1,<1.25, installed: 1.22]
    - responses [required: >=0.9.0, installed: 0.10.4]
      - requests [required: >=2.0, installed: 2.20.0]
        - certifi [required: >=2017.4.17, installed: 2018.11.29]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.8, installed: 2.7]
        - urllib3 [required: >=1.21.1,<1.25, installed: 1.22]
      - six [required: Any, installed: 1.10.0]
    - six [required: >1.9, installed: 1.10.0]
    - werkzeug [required: Any, installed: 0.14.1]
    - xmltodict [required: Any, installed: 0.10.2]
  - nose [required: ==1.3.7, installed: 1.3.7]
  - psutil [required: ==5.4.3, installed: 5.4.3]
  - pyopenssl [required: ==17.5.0, installed: 17.5.0]
    - cryptography [required: >=2.1.4, installed: 2.4.2]
      - asn1crypto [required: >=0.21.0, installed: 0.24.0]
      - cffi [required: >=1.7,!=1.11.3, installed: 1.11.5]
        - pycparser [required: Any, installed: 2.19]
      - idna [required: >=2.1, installed: 2.7]
      - six [required: >=1.4.1, installed: 1.10.0]
    - six [required: >=1.5.2, installed: 1.10.0]
  - python-coveralls [required: ==2.7.0, installed: 2.7.0]
    - coverage [required: ==4.0.3, installed: 4.0.3]
    - PyYAML [required: Any, installed: 4.2b4]
    - requests [required: Any, installed: 2.20.0]
      - certifi [required: >=2017.4.17, installed: 2018.11.29]
      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<2.8, installed: 2.7]
      - urllib3 [required: >=1.21.1,<1.25, installed: 1.22]
    - sh [required: Any, installed: 1.12.14]
    - six [required: Any, installed: 1.10.0]
  - pyyaml [required: ==4.2b4, installed: 4.2b4]
  - requests [required: ==2.20.0, installed: 2.20.0]
    - certifi [required: >=2017.4.17, installed: 2018.11.29]
    - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
    - idna [required: >=2.5,<2.8, installed: 2.7]
    - urllib3 [required: >=1.21.1,<1.25, installed: 1.22]
  - requests-aws4auth [required: ==0.9, installed: 0.9]
    - requests [required: Any, installed: 2.20.0]
      - certifi [required: >=2017.4.17, installed: 2018.11.29]
      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<2.8, installed: 2.7]
      - urllib3 [required: >=1.21.1,<1.25, installed: 1.22]
  - six [required: ==1.10.0, installed: 1.10.0]
  - subprocess32-ext [required: ==3.2.8.2, installed: 3.2.8.2]
  - xmltodict [required: ==0.10.2, installed: 0.10.2]


```

**Comment by rednaks** [comment]
I'm having the same issue here, did you find out how to fix it ? (I didn't try build from sources yet)

**Comment by michaeljohnbennett** [comment]
YMMV but this installed finally for me:

``` pip install --upgrade --ignore-install localstack```

I did also use miniconda and even its pyyaml installer was a distutils.
```
> pip show pyyaml                               
Name: PyYAML
Version: 4.2b4
Summary: YAML parser and emitter for Python
Home-page: https://github.com/yaml/pyyaml
Author: Kirill Simonov
Author-email: xi@resolvent.net
License: MIT
Location: /Users/xxxx/miniconda3/lib/python3.7/site-packages
Requires:
Required-by: python-coveralls, pyaml, flask-swagger, awscli, anaconda-client
```

I deleted the `yaml` folder the `PyYaml....egg.info` file and reinstalled, it still complained but I didn't get an error at the end.

**Comment by rednaks** [comment]
setting a fixed version to >=3.14 solved the issue for me, I'll PR and see if they accept it.
